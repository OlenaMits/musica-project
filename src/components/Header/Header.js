import React from "react";
import { useSelector } from "react-redux";
import { ReactComponent as Logo } from './images/musician-guitar-player.svg';
import { ReactComponent as Favorite } from './images/favorite.svg';
import { ReactComponent as Cart } from './images/cart.svg';
import { Link } from "react-router-dom";

import "./Header.scss";

const Header = () => {
    const favorites = useSelector((state) => state.favorites);
    const cart = useSelector((state) => state.cart);

    return(
        <div className="header" data-testid="app-header">
            <div className="container">
                <div className="header-logo">
                    <Link to="/">
                        <Logo/>
                        <p>Musica</p>
                    </Link>
                </div>
                <div className="header-btns">
                    <div className="favorite">
                        <Link to="favorite">
                            <Favorite/>
                        </Link>
                        <div className="counter">{favorites.length || ""}</div>
                    </div>
                    <div className="cart">
                        <Link to="cart">
                            <Cart/>
                        </Link>
                        <div className="counter">{cart.length || ""}</div>
                    </div>
                </div>
            </div>
        </div>
    )
};

export default Header;