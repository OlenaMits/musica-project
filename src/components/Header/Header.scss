.header {
    position: sticky;
    width: 100%;
    height: auto;
    top: 0;
    background-color: #aedaa6;
    opacity: 0.8;
    z-index: 1;

    .container {
        width: 90%;
        padding: 15px;
        max-width: 1200px;
        margin: auto;
        display: flex;
        justify-content: space-between;
        align-items: center;
        color: #000;
        font-weight: 500;
        font-size: 20px;

        .header-logo {
            a {
                color: #000;
                text-decoration: none;
                display: flex;
                align-items: center;

                p {
                    margin: 0;
                }
            }
        }

        .header-btns {
            display: flex;
            justify-content: space-between;
            align-items: baseline;
            gap: 15px;


            .favorite {
                cursor: pointer;
                display: flex;
                justify-content: space-between;
                align-items: center;
                gap: 5px;
            }

            .cart {
                cursor: pointer;
                display: flex;
                justify-content: space-between;
                align-items: center;
                gap: 5px;
            }

            .counter {
                width: 10px;
                font-size: 16px;
            }
        }
    }
}