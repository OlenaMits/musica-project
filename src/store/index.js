import { combineReducers, createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import { loadState, saveState } from "../services/localStorageService"; 
import goodsReducer from "./reducers/goodsReducer";
import favoritesReducer from "./reducers/favoriteReducer";
import cartReducer from "./reducers/cartReducer";
import modalReducer from "./reducers/modalReducer";

const rootReducer = combineReducers({
    goods: goodsReducer,
    favorites: favoritesReducer,
    cart: cartReducer,
    modal: modalReducer,
});

const persistedState = loadState();

const composedEnhancer = composeWithDevTools(
    applyMiddleware(thunk)
);

const store = createStore(rootReducer, persistedState, composedEnhancer);
store.subscribe(() => {
    const currentState = store.getState();
    saveState(currentState);
});

export default store;