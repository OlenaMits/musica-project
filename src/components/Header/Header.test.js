import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

import Header from "./Header";
import { Provider } from 'react-redux';
import store from "../../store/index";

describe('Card tests', () => {
    it('should have text Musica', () => {
        render(
            <Provider store={store}>
                <Header/>
            </Provider>
        , { wrapper: BrowserRouter });
        expect(screen.getByText(/Musica/)).toBeTruthy();
    });

    it('should match snapshot', () => {
        const {asFragment} = render(
            <Provider store={store}>
                <Header/>
            </Provider>
        , { wrapper: BrowserRouter });

        const element = screen.getByTestId('app-header');

        expect(element).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
  
    it('should find header', async () => {
        render(
            <Provider store={store}>
                <Header/>
            </Provider>
        , { wrapper: BrowserRouter });

        const element = screen.getByTestId('app-header');

        expect(element).toBeInTheDocument();
    });
});