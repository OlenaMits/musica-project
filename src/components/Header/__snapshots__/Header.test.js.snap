// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="header"
    data-testid="app-header"
  >
    <div
      class="container"
    >
      <div
        class="header-logo"
      >
        <a
          href="/"
        >
          <svg>
            musician-guitar-player.svg
          </svg>
          <p>
            Musica
          </p>
        </a>
      </div>
      <div
        class="header-btns"
      >
        <div
          class="favorite"
        >
          <a
            href="/favorite"
          >
            <svg>
              favorite.svg
            </svg>
          </a>
          <div
            class="counter"
          />
        </div>
        <div
          class="cart"
        >
          <a
            href="/cart"
          >
            <svg>
              cart.svg
            </svg>
          </a>
          <div
            class="counter"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
